calcrule_social_protection/benefit_plan_strategy.py

```diff
@@ -0,0 +1,71 @@
+from core.abs_calculation_rule import AbsCalculationRule
+
+from calcrule_social_protection.interfaces import (
+    GroupBenefitPlanInterface,
+    IndividualBenefitPlanInterface
+)
+
+
+class BenefitPackageStrategy(AbsCalculationRule):
+
+    CLASS_NAME_CHECK = 'PaymentPlan'
+
+    @classmethod
+    def run_calculation_rules(cls, sender, payment_plan, user, context, **kwargs):
+        return cls.calculate_if_active_for_object(payment_plan, **kwargs)
+
+    @classmethod
+    def calculate_if_active_for_object(cls, payment_plan, **kwargs):
+        if cls.active_for_object(payment_plan):
+            return cls.calculate(payment_plan, **kwargs)
+
+    @classmethod
+    def active_for_object(cls, payment_plan):
+        return cls.check_calculation(payment_plan)
+
+    @classmethod
+    def get_linked_class(cls, sender, class_name, **kwargs):
+        return ["Calculation"]
+
+    @classmethod
+    def get_parameters(cls, sender, class_name, instance, **kwargs):
+        rule_details = cls.get_rule_details(sender=sender, class_name=class_name)
+        if rule_details:
+            if instance.__class__.__name__ == cls.CLASS_NAME_CHECK:
+                if cls.check_calculation(payment_plan=instance):
+                    return rule_details["parameters"] if "parameters" in rule_details else []
+
+    @classmethod
+    def run_convert(cls, payment_plan, **kwargs):
+        return cls.convert(payment_plan=payment_plan, **kwargs)
+
+    @classmethod
+    def check_calculation(cls, payment_plan, **kwargs):
+        benefit_plan_type = payment_plan.benefit_plan.type
+        if benefit_plan_type == GroupBenefitPlanInterface.TYPE:
+            return GroupBenefitPlanInterface.check_calculation(cls, payment_plan)
+        if benefit_plan_type == IndividualBenefitPlanInterface.TYPE:
+            return IndividualBenefitPlanInterface.check_calculation(cls, payment_plan)
```

Comment:
There should be single check for the interface. Both GroupBenefitPlanInterface and IndividualBenefitPlanInterface are interfaces implementation. In that case every new implementation of the interface would require additional if statement with is not optimal (e.g. in terms of open-close principle). Instead you should: 
1. Define single interface (for example named BenefitPlanStrategy)
2. Add implementations for GroupBenefitPlanInterface and IndividualBenefitPlanInterface 
3. Change functions in the benefit_plan_strategy to check if interface is used (it shouldn't be done by checking instance or inheritance, rather the content, I think Pydantic or other libraries are able do do this. 
4. If it's interface then you know what functions can be executed. 

You're using this if/else in several methods across this calculation rule. It's be more efficient to determine once (ideally before calc rule is executed) what actual interface implementation is used. It should be done once and all remaining logic should rely on abstraction. If it's not possible to pass class implementing interface with constructor or parameter (because classmethods are used) you could consider creating another layer with strategies registry and call them i.e. as: 
```BenefitPlanStrategyStorage(payment_plan).check_calculation()```